jsonschema/src/compilation/options.rs

```diff
@@ -601,6 +651,34 @@ impl fmt::Debug for CompilationOptions {
     }
 }
 
+/// Define a custom keyword validation
+///
+/// Currently supports validation functions.
+/// Schema based custom validation is a potential future addition.
+#[derive(Clone)]
+pub enum CustomKeywordDefinition {
+    /// Validate [instance](serde_json::Value) according to a custom specification
+    ///
+    /// A custom keyword validator may be used when a validation that cannot, or
+    /// cannot be be easily or efficiently expressed in JSON schema.
+    ///
+    /// The custom validation is applied in addition to the JSON schema validation.
+    Validator {
+        /// Validate an instance returning any and all detected validation errors
+        validate: CustomValidateFn,
+        /// Determine if an instance is valid
+        is_valid: CustomIsValidFn,
```

Comment:
Since the distinction between `validate` and `is_valid` only exists to improve the performance of `is_valid`, we could allow the user to omit this by making this an `Option<CustomIsValidFn>`, and have the `is_valid()` implementation of the `Validate` trait have a default case if this is missing that calls `validate` and returns if any errors are present. This would allow the user to make the tradeoff of losing out on `is_valid()` perf a bit but avoiding duplication of the validation logic.

Reply:
Ah I see @Stranger6667 noted this in his comment just above, wrt the trait default implementation